alchemist-graphql/src/jvmTest/kotlin/it/unibo/alchemist/boundary/TestingEnvironments.kt

```diff
@@ -15,14 +15,15 @@ import it.unibo.alchemist.model.EuclideanEnvironment
 import it.unibo.alchemist.model.Position
 import it.unibo.alchemist.model.geometry.Vector
 import it.unibo.alchemist.test.loadYamlSimulation
+import it.unibo.alchemist.test.startSimulation
 import java.io.File
 
 object TestingEnvironments {
     fun <T, P> graphqlTestEnvironmnets(): Set<EuclideanEnvironment<T, P>> where P : Position<P>, P : Vector<P> =
         this::class.java.classLoader.getResource("yaml")?.path?.let { path ->
             File(path).listFiles()?.map {
                 loadYamlSimulation<T, P>("yaml/${it.name}")
-            }?.toSet()
+            }?.toSet()?.onEach { it.startSimulation() }
```

Comment:
maybe using a sequence and an executor? Otherwise it preloads everything in memory...

Reply:
I've changed the code in order to use a sequence. I've tried to use executors through java `ExecutorService` but then the `addNode` method in [environment test](https://github.com/AlchemistSimulator/Alchemist/pull/2706/files#diff-f5fb2b229c40eb15a21c4707cbe855f9cbf1278967c527dd1e3569eb4d4c7d3eR40) fails, because the method should be called inside the simulation thread, and if I try to use the `Simulation.schedule` method, an exception is thrown due to the termination of the simulation. Maybe i am messing up with threads and the simulation launching procedure.